### PR TITLE
feat/FOROME-811: add disabling apply button on filter refiner

### PR DESCRIPTION
**Pull request by QSdmitrioul**
https://quantori.atlassian.net/browse/FOROME-811

Realization appearing not clear. After adding architecture solution for redoing actions, it can be improved.

### CODE REVIEW

File: src/store/filter/filter.store.ts [hunk 1]
```diff
@@ -136,6 +137,10 @@ export class FilterStore {
     return this._presetModifiedState === PresetModifiedState.Modified
   }
 
+  public get isChanging(): boolean {
+    return this._isChanging
```
Comment (AndreyUstyumenko) [inline src/store/filter/filter.store.ts]: just make isChanging public

File: src/pages/filter/ui/panels/function-panel/components/gene-region/gene-region.tsx [hunk 2]
```diff
@@ -65,6 +66,22 @@ export const GeneRegion = observer(() => {
...
 
+  const toggleNotMode = () => {
+    geneRegionStore.setCurrentMode(ModeTypes.Not)
+    filterStore.setChanging(true)
+  }
+
+  const onChangeInput = (e: any) => {
+    geneRegionStore.setLocusValue(e.target.value)
+    validateValue(e.target.value)
+    filterStore.setChanging(true)
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/panels/function-panel/components/gene-region/gene-region.tsx]: you must check if there is any value or the value is not the same as initial. Relates to all alike cases

File: src/pages/filter/ui/query-builder/right-refiner-column/query-selected.tsx [hunk 3]
```diff
@@ -44,6 +45,12 @@ export const QuerySelected = observer((): ReactElement => {
     filterStore.clearConditions()
   }
 
+  const isDisabledApplyButton =
+    isConditionsFetching ||
+    filteredVariantsCount == null ||
+    !conditions.length ||
+    isChanging
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/query-builder/right-refiner-column/query-selected.tsx]: Looks like isChanging should be opposite isn't it? The button should be disabled when isChanging is false?

File: src/store/filter/filter.store.ts [hunk 4]
```diff
@@ -33,6 +33,7 @@ export class FilterStore {
   private _attributeNameToAdd: string = ''
   private _presetModifiedState: PresetModifiedState =
     PresetModifiedState.NotPreset
+  private _isChanging = false
```
Comment (AndreyUstyumenko) [inline src/store/filter/filter.store.ts]: In form development it is common to name such properties 'isDirty'. When the form is touched (some value is enterd or changed) it is considered to be 'dirty'. Or simply 'isTouched'.
I propose to call it 'isFilterTouched'

File: src/pages/filter/ui/query-builder/right-refiner-column/query-selected.tsx (hunk 3 above)
Comment (QSdmitrioul) [reply src/pages/filter/ui/query-builder/right-refiner-column/query-selected.tsx]: No, it is not. It is apply button for filter, so we can apply only when we don't have any unsaved changes.

File: src/store/filter/filter.store.ts (hunk 4 above)
Comment (QSdmitrioul) [reply src/store/filter/filter.store.ts]: Done

File: src/store/filter/filter.store.ts (hunk 1 above)
Comment (QSdmitrioul) [reply src/store/filter/filter.store.ts]: Done

File: src/pages/filter/ui/panels/function-panel/components/gene-region/gene-region.tsx (hunk 2 above)
Comment (QSdmitrioul) [reply src/pages/filter/ui/panels/function-panel/components/gene-region/gene-region.tsx]: Done

**Review by AndreyUstyumenko — APPROVED** [review]

**Review by Lapk1n — APPROVED** [review]

**Pull request merged**